test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionMetadataProviderTest.cs

```diff
@@ -148,6 +148,29 @@ public void FromStringExpression_SetsContainerAsExpected()
             Assert.Same(myModel, metadata.Container.Model);
         }
 
+        // This is a special case that's basically only possible with Razor Pages.
```

Comment:
> Why is the private property case only possible with Razor Pages?

You're more likely to think it would work.

I'm happy to try a few other things